validator/sawtooth_validator/gossip/signature_verifier.py

```diff
@@ -93,6 +94,16 @@ def validate_batch(batch):
                              batch.header_signature,
                              txn.header_signature)
                 valid = False
+                return valid
+
+            # verify the payload field matches the header
+            txn_payload_sha512 = hashlib.sha512(txn.payload).hexdigest()
+            if txn_payload_sha512 != txn_header.payload_sha512:
+                LOGGER.debug("the payload field doesn't match the header"
+                             "for txn: %s", txn.header_signature)
+                valid = False
+                return valid
```

Comment:
Also, a nit, but just `return False`.  Don't bother with the assignment to local.